### PR TITLE
Feature/wv 2579

### PR DESCRIPTION
**Pull request by maximilianmikus**

### CODE REVIEW

File: tests/behat/behat-features/src/JsonApiContext.php [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    }
+    $this->contentpoolBaseUrl = getenv('CONTENTPOOL_BASE_URL');
+
+    $client = Client::createWithConfig([]);
+
+    $this->jsonApiClient = new JsonApiClient($client);
+  }
+
+  /**
+   * @Then I request an article with the uuid :uuid
```
Comment (fago) [inline tests/behat/behat-features/src/JsonApiContext.php]: as sstep definitions are global, we should always bring in json_api here:

>Then I request ... via json-api zB?

File: tests/behat/behat-features/src/JsonApiContext.php [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+    $requestBuilder
+      ->setProtocolVersion("1.1")
+      ->setMethod("GET")
+      ->setUri("{$this->contentpoolBaseUrl}/jsonapi/node/article/$uuid")
+      ->setHeader("Accept-Charset", "utf-8");
+
+    $request = $requestBuilder->getRequest();
+
+    $response = $this->jsonApiClient->sendRequest($request);
```
Comment (fago) [inline tests/behat/behat-features/src/JsonApiContext.php]: Let's write the response in $this->response and move the expection to a separate step like
"And the json-api response is successful" - this is the same way out behat integration works, e.g. drush command "and the drush output is ..."

File: tests/behat/behat-features/src/JsonApiContext.php [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    // Instantiate an empty PSR-7 request.
+    $consumer_id = $this->getSession()->getDriver()->evaluateScript('jQuery("td:contains(\'BEHAT Consumer\')").first().prev().text()');
+    if (empty($consumer_id)) {
+      throw new ExpectationException('Unable to save consumer id.', $this->getSession());
+    }
+    $this->consumerId = $consumer_id;
+  }
+
+  /**
+   * @Then I request an article with the uuid :uuid and included :fields
```
Comment (fago) [inline tests/behat/behat-features/src/JsonApiContext.php]: via json-api

File: tests/behat/behat-features/src/JsonApiContext.php [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+    $response = $this->jsonApiClient->sendRequest($request);
+
+    if (!$response->isSuccessful() || !$response->hasDocument()) {
+      throw new ExpectationException('Request to fetch an article was not successful or has no results', $this->getSession());
+    }
+
+  }
+
+  /**
+   * @Then I save the consumer id
```
Comment (fago) [inline tests/behat/behat-features/src/JsonApiContext.php]: would be nice to provide more context, e.g. "Then I save the consumer id for oauth authentication"

File: tests/behat/behat-features/src/JsonApiContext.php [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      "grant_type" => "password",
+    ];
+
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, "{$this->contentpoolBaseUrl}/oauth/token");
+    curl_setopt($ch, CURLOPT_POST, TRUE);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    /* curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, FALSE); */
+    curl_setopt($ch, CURLOPT_POSTFIELDS, $config);
+    $data = curl_exec($ch);
```
Comment (fago) [inline tests/behat/behat-features/src/JsonApiContext.php:225]: you could have used guzzle, but nevermind

**Review by fago — APPROVED** [review]

File: tests/behat/behat-features/src/JsonApiContext.php (hunk 5 above)
Comment (maximilianmikus) [reply tests/behat/behat-features/src/JsonApiContext.php:225]: tried to use guzzle for oauth2 in the test, but had problems making the yang jsonapi client work with it

**Review by fago — APPROVED** [review]

**Pull request merged**